### DISCUSSION TITLE
Nested Filters?

### DISCUSSION THREAD

**Discussion by kushfest** [anvilistas/tabulator#45] [Q&A]
I'm trying to create a search bar that can search across multiple fields. I see in the tabulator docs that I can search multiple filters with OR logic by passing an array of filters as the second search object.

```
able.setFilter([
    {field:"age", type:">", value:52}, //filter by age greater than 52
    [
        {field:"height", type:"<", value:142}, //with a height of less than 142
        {field:"name", type:"=", value:"steve"}, //or a name of steve
    ]
]);
```

Is there a way to do this in anvil?

**Comment by s-cork** [comment]
what happens when you just do?

`self.tabulator_1.set_filter([...])`

set_filter will call the underlying Javascript method.

**Comment by kushfest** [comment]
I feel like I'm missing something obvious (and probably basic python), but I get ```TypeError: unhashable type: 'dict'``` from this code:
```
self.locations_grid.set_filter([
        {"like",{"name": self.search_text.text}},
        {"like",{"phone": self.search_text.text}},
        {"like",{"email": self.search_text.text}},
        {"like",{"website": self.search_text.text}},
        {"like",{"instagram": self.search_text.text}},
        {"like",{"address": self.search_text.text}},
        {"like",{"city": self.search_text.text}},
        {"like",{"state": self.search_text.text}},
        {"like",{"notes": self.search_text.text}}
      ])
```

My intention is to use the internal "like" filter across multiple fields, so that if any of the filters return results, they are displayed.

**Reply by s-cork** [reply]
The comma after like makes it a set not a dict.

**Comment by kushfest** [comment]
Thank you. That was obvious thing I was doing wrong. I still haven't gotten this working, but will keep at it.
